lib/rdoc/generator/template/darkfish/_head.rhtml

```diff
@@ -25,6 +25,12 @@
   <%- end -%>
 <%- end -%>
 
+<%- if @options.canonical_root -%>
+  <% context = klass if defined?(klass) %>
+  <% context = file if defined?(file) %>
+  <link rel="canonical" href="<%= canonical_url(context) %>">
```

Comment:
Can this be included in the if statements above? Then we don't need the context conditions separately.

Reply:
I've changed the implementation to use `current` instead of `klass` and `file` to simplify.
If I'd use the `if` statement I would need to add an `else` as well, resulting in adding the tag 4 times.